File: backend/src/Statera.Api/Contracts/Forecast.cs
// backend/src/Statera.Api/Contracts/Forecast.cs
namespace Statera.Api.Contracts;

// ── Demand ──────────────────────────────────────────────────────────────────
public record DemandForecastRequest(Guid FacilityId, Guid UnitId, DateOnly From, DateOnly To);
public record DemandPoint(DateOnly Date, int RequiredHeads, int HistoricalAvgHeads, double Confidence);
public record DemandForecastResponse(IEnumerable<DemandPoint> Points);

// ── Burnout ──────────────────────────────────────────────────────────────────
public record BurnoutRiskRequest(Guid FacilityId, DateOnly From, DateOnly To);
public record BurnoutPoint(Guid StaffId, string StaffName, double RiskScore,
    double HoursLast4Wk, int ConsecutiveDays, int ShiftsLast4Wk);
public record BurnoutRiskResponse(IEnumerable<BurnoutPoint> Points);

// ── What-if ──────────────────────────────────────────────────────────────────
public record WhatIfScenarioRequest(Guid FacilityId, Guid UnitId, string Scenario,
    DateOnly From, DateOnly To);
public record WhatIfEffect(string Day, double CurrentCoverage, double ProjectedCoverage);
public record WhatIfResponse(string Scenario, double CurrentAvgCoverage,
    double ProjectedAvgCoverage, int GapDays, bool OvertimeRisk,
    IEnumerable<WhatIfEffect> DayEffects, IEnumerable<string> Recommendations);

// ── AI Insights ──────────────────────────────────────────────────────────────
public record ForecastInsightsRequest(Guid FacilityId, Guid? UnitId, DateOnly From, DateOnly To);
public record AiInsight(string Title, string Body, string Severity); // info | warning | critical
public record ForecastSummary(double AvgCoverageRate, int UnderstaffedDays,
    int OvertimeDays, double PeakDemandDay, string PeakDayOfWeek);
public record ForecastInsightsResponse(ForecastSummary Summary,
    IEnumerable<AiInsight> Insights, string? TokensUsed);

File: frontend/statera-web/src/api/forecast.ts
// src/api/forecast.ts
import api from "./axios";

export interface DemandPoint {
  date: string;
  requiredHeads: number;
  historicalAvgHeads: number;
  confidence: number;
}

export interface DemandForecastResponse {
  points: DemandPoint[];
}

export interface BurnoutPoint {
  staffId: string;
  staffName: string;
  riskScore: number;
  hoursLast4Wk: number;
  consecutiveDays: number;
  shiftsLast4Wk: number;
}

export interface BurnoutRiskResponse {
  points: BurnoutPoint[];
}

export interface WhatIfEffect {
  day: string;
  currentCoverage: number;
  projectedCoverage: number;
}

export interface WhatIfResponse {
  scenario: string;
  currentAvgCoverage: number;
  projectedAvgCoverage: number;
  gapDays: number;
  overtimeRisk: boolean;
  dayEffects: WhatIfEffect[];
  recommendations: string[];
}

export interface AiInsight {
  title: string;
  body: string;
  severity: "info" | "warning" | "critical";
}

export interface ForecastSummary {
  avgCoverageRate: number;
  understaffedDays: number;
  overtimeDays: number;
  peakDemandDay: number;
  peakDayOfWeek: string;
}

export interface ForecastInsightsResponse {
  summary: ForecastSummary;
  insights: AiInsight[];
  tokensUsed?: string | null;
}

export const forecastApi = {
  getDemand: (facilityId: string, unitId: string, from: string, to: string) =>
    api.post<DemandForecastResponse>("/forecast/demand", { facilityId, unitId, from, to }),

  getBurnout: (facilityId: string, from: string, to: string) =>
    api.post<BurnoutRiskResponse>("/forecast/burnout", { facilityId, from, to }),

  getWhatIf: (facilityId: string, unitId: string, scenario: string, from: string, to: string) =>
    api.post<WhatIfResponse>("/forecast/whatif", { facilityId, unitId, scenario, from, to }),

  getInsights: (facilityId: string, unitId: string | null, from: string, to: string) =>
    api.post<ForecastInsightsResponse>("/forecast/insights", { facilityId, unitId, from, to }),
};
